app/services/replacement_parameters_service.rb

```diff
@@ -45,9 +45,13 @@ def you_or_your
   # this should only be called when we aren't saving the output to our database
   # for example, when adding a link to send to the client only
   def sensitive_replacements
-    {
-      "Link.E-signature": client.login_link
-    }
+    if body.match(/<<\s*Link\.E-signature\s*>>/i)
```

Comment:
I'm surprised you didn't need to do `\-` instead of `-` but if it works, fine with me!